src/app.js

```diff
@@ -0,0 +1,275 @@
+    let formatNameAttr
+    if (Number.isInteger(this.#mediaInfo.frameRate)) {
+      this.#tcMultiplier = 1
+      this.#tcDenominator = this.#mediaInfo.frameRate
+      this.#initialOffset = this.#mediaInfo.frameRate * 3600 // davinci default
+      formatNameAttr =
+        'FFVideoFormat1080p' + this.#mediaInfo.frameRate.toString()
+    } else {
+      // for videos that use drop-frame e.g. 29.97 fps (30000/1001 fps)
+      this.#tcMultiplier = 1001 // e.g. 29.97 * 1001 = 30000
+      this.#tcDenominator = this.#mediaInfo.frameRate * this.#tcMultiplier
+      this.#initialOffset = this.#mediaInfo.frameRate * 3603.6
+      formatNameAttr =
+        'FFVideoFormat1080p' +
+        this.#mediaInfo.frameRate
+          .toFixed(2)
+          .toString()
+          .replace('.', '')
+    }
+    let formatFrameDurationAttr = this.toTimecodeStr(1) // 1/frameRate
+    let formatWidthAttr = this.#mediaInfo.width
+    let formatHeightAttr = this.#mediaInfo.height
+    if (this.#mediaInfo.hasVideo) {
+      this.#startStr = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE fcpxml><fcpxml version="1.9"><resources><format id="r0" name="${formatNameAttr}" width="1920" height="1080" frameDuration="${formatFrameDurationAttr}"/><format id="r1" name="FFVideoFormatRateUndefined" width="${formatWidthAttr}" height="${formatHeightAttr}" frameDuration="${formatFrameDurationAttr}"/>`
+      this.#endStr = `</spine></sequence></project></event></library></fcpxml>`
+      this.#currentAssetID = 2 // 0 and 1 are reserved for <format>
+    } else {
+      this.#startStr = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE fcpxml><fcpxml version="1.9"><resources><format id="r0" name="FFVideoFormat1080p24" width="1920" height="1080" frameDuration="1/24s"/>`
+      this.#endStr = `</gap></spine></sequence></project></event></library></fcpxml>`
+      this.#currentAssetID = 1 // 0 is reserved for <format>
+    }
+    this.#assetList = []
+    this.#eventStr = `</resources><library><event name="Timeline 1"><project name="Timeline 1">`
+    this.#sequenceStr = ''
+    this.#clipList = []
+    this.#currentOffset = this.#initialOffset
+  }
+  gcd (a, b) {
+    let temp
+    while (b !== 0) {
+      temp = a % b
+      a = b
+      b = temp
+    }
+    return a
+  }
+  toTimecodeStr (frameNumber) {
+    let tcNumerator = frameNumber * this.#tcMultiplier
+    let divisor = this.gcd(tcNumerator, this.#tcDenominator)
+    return (
+      (tcNumerator / divisor).toString() +
+      '/' +
+      (this.#tcDenominator / divisor).toString() +
+      's'
+    )
+  }
+  addClip (startFrame, endFrame) {
+    let asset = ''
+    let clip = ''
+    let assetIDAttr = 'r' + this.#currentAssetID.toString()
+    let assetNameAttr = this.#mediaInfo.fileName
+    let assetDurationAttr = this.toTimecodeStr(this.#mediaInfo.frameCount)
+    let clipStartAttr = this.toTimecodeStr(startFrame)
+    let clipDuration = endFrame - startFrame
+    let clipDurationAttr = this.toTimecodeStr(clipDuration)
```

Comment:
minor style suggestion: consider using const instead of let if the variable does not need to be mutated.

Reply:
StandardJS states that all variables should be camelCase- I was thinking we should reserve the const keyword to serve the same purpose as SCREEN_CASE since it's highlighted differently in the editor. Basically, use const if you would use SCREEN_CASE for that variable, if not, don't use const.